hamilton/plugins/h_pyarrow.py

```diff
@@ -0,0 +1,39 @@
+from typing import Any, Type
+
+import pyarrow
+from pyarrow.interchange import from_dataframe
+
+from hamilton.lifecycle.api import ResultBuilder
+
+
+class PyarrowTableResult(ResultBuilder):
+    """Add this result builder to a materializer's `combine` statement to convert your dataframe
+    object to a pyarrow representation and make it compatible with pyarrow DataSavers.
+
+    It implicitly support input_type == Any, but it expects dataframe objects implementing
+    the dataframe interchange protocol: ref: https://arrow.apache.org/docs/python/interchange_protocol.html
+    for example:
+    - pandas
+    - polars
+    - dask
+    - vaex
+    - ibis
+    - duckdb results
```

Comment:
it would be nice to be stricter on types -- but I suppose this is fine.

Reply:
```python
def input_types(self) -> List[Type[Type]]:
    """Gives the applicable types to this result builder.
    This is optional for backwards compatibility, but is recommended.

    :return: A list of types that this can apply to.
    """
    _types = []
    try:
       import ...
   except ...
    return _types
```

Reply:
I couldn't figure a way to annotate type without a series of `try/except` statements. After feedback on the dlt extension, I thought this pattern undesirable, so I opted to not include `input_types()` if it was to return `Any`